services/management/dbaas/components_service.go

```diff
@@ -192,6 +196,101 @@ func (c componentsService) ChangePXCComponents(ctx context.Context, req *dbaasv1
+			err: err,
+		}
+		return
+	}
+	responseCh <- checkResponse{
+		kuberentesClusterName: kuberentesCluster.KubernetesClusterName,
+		pxcOperatorVersion:    resp.Operators.Xtradb.Version,
+		psmdbOperatorVersion:  resp.Operators.Psmdb.Version,
+	}
+}
+
+func doesOperatorNeedUpdate(installedOperatorVersion, latestOperatorForInstalledPMM *goversion.Version) (availableOperatorVersion string) {
+	if latestOperatorForInstalledPMM.GreaterThan(installedOperatorVersion) {
+		return latestOperatorForInstalledPMM.String()
+	}
+	return ""
+}
+
+func (c componentsService) CheckForOperatorUpdate(ctx context.Context, req *dbaasv1beta1.CheckForOperatorUpdateRequest) (*dbaasv1beta1.CheckForOperatorUpdateResponse, error) {
+	if pmmversion.PMMVersion == "" {
+		return nil, status.Error(codes.Internal, "failed to get current PMM version")
+	}
+
+	// List operator versions in all kuberenetes clusters.
+	clusters, err := models.FindAllKubernetesClusters(c.db.Querier)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	// And get operators version from all of them.
+	responseCh := make(chan checkResponse, len(clusters))
+	go func() {
+		wg := &sync.WaitGroup{}
+		wg.Add(len(clusters))
+		for _, cluster := range clusters {
+			k8sCluster := cluster
+			go c.getInstalledOperatorsVersion(ctx, wg, responseCh, k8sCluster)
+		}
+		wg.Wait()
+		close(responseCh)
+	}()
+
+	// Meanwhile, get latest operators version compatible with installed PMM.
+	pmmVersionParts := strings.Split(pmmversion.PMMVersion, "-") // handle development builds, for example: 2.18.0-27-g1e5f59d-dirty
+	latestPXCOperatorForInstalledPMM, latestPSMDBOperatorForInstalledPMM, err := c.versionServiceClient.GetLatestOperatorVersion(ctx, pmmVersionParts[0])
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	if latestPXCOperatorForInstalledPMM == nil || latestPSMDBOperatorForInstalledPMM == nil {
+		return nil, status.Error(codes.Internal, "latest versions are of invalid values")
+	}
+
+	resp := &dbaasv1beta1.CheckForOperatorUpdateResponse{
+		ClusterToComponents: make(map[string]*dbaasv1beta1.ComponentsUpdateInformation),
+	}
+	// Some of the requests to kuberenetes clusters for getting operators versions should be done.
+	// Go through them and decide what operator needs update.
+	for operatorsVersion := range responseCh {
+		if operatorsVersion.err != nil {
+			c.l.Error(operatorsVersion.err)
+			resp.ClusterToComponents[operatorsVersion.kuberentesClusterName] = &dbaasv1beta1.ComponentsUpdateInformation{}
+			continue
+		}
+		installedPXCOperatorVersion, pxcErr := goversion.NewVersion(operatorsVersion.pxcOperatorVersion)
+		installedPSMDBOperatorVersion, psmdbErr := goversion.NewVersion(operatorsVersion.psmdbOperatorVersion)
+		if pxcErr != nil || psmdbErr != nil {
+			c.l.Errorf("failed to parse versions: pxc operator: %q, psmdb operator: %q", operatorsVersion.pxcOperatorVersion, operatorsVersion.psmdbOperatorVersion)
+			continue
+		}
```

Comment:
pxc or psmdb might not be installed, so we should just ignore update in that case.

Reply:
done